a11y/Dockerfile

```
@@ -0,0 +1,43 @@
+FROM node
+
+LABEL maintainer="daniel.walford@ons.gov.uk"
+
+# Explicitly declare npm to root (docker runs as root anyway) allows global package installation for npm, needed for @axe-core/cli
+RUN npm --global config set user root
+
+# We need wget to set up the Personal Package Archive (PPA) and unzip to install the Chromedriver
+RUN apt update
+RUN apt install -y wget unzip
+
+# Set up the Chrome PPA
+RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -
+RUN echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list
```

Try to minimise the number of `RUN` instructions in the dockerfile. Each RUN creates a new image layer. See (https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers)
```suggestion
RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
   &&  echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list
```

---

Thanks, tried to group sets of related tasks under single RUN commands, however, as spotted there are 2 which could be grouped better. Just made an update to reduce those two.

---

It's still not ideal to have so many layers created in a dockerfile, we have 8 layers in this one file. Every time there is a `RUN` in a Dockerfile it creates a new image layer and each layer gets pulled from the repository every time a container is created (ignoring caches which does massively speed things up). If you can, reduce the number of `RUN` commands in the file to the absolute minimum. For example, look at how openjdk does[ oodles of steps](https://github.com/docker-library/openjdk/blob/master/Dockerfile-adopt-debian.template#L43) under one `RUN` command in their example, even adding comments between the steps.

---

I have reduced it down to 3 now (was initially 10) I could in theory put them all into one.
- apt installations
- Chromedriver setup
- npm installations